### ISSUE TITLE
Barcode Text not rendered in Viewer output

### ISSUE THREAD

**Issue by primo-ppcg** [BinaryKits/BinaryKits.Zpl#119]
Using:

.NET Framework 4.7.2
BinaryKits.Zpl.Label 3.1.0
BinaryKits.Zpl.Viewer 1.1.1

All other dependencies up to date.

Method for creating label preview:

```csharp
DialogResult result = this.openFileDialog.ShowDialog();
if(result == DialogResult.OK) {
    string content = File.ReadAllText(this.openFileDialog.FileName);

    IPrinterStorage printerStorage = new PrinterStorage();
    ZplElementDrawer drawer = new ZplElementDrawer(printerStorage);

    ZplAnalyzer analyzer = new ZplAnalyzer(printerStorage);
    AnalyzeInfo analyzeInfo = analyzer.Analyze(content);

    foreach(LabelInfo labelInfo in analyzeInfo.LabelInfos) {
        byte[] labelData = drawer.Draw(labelInfo.ZplElements);
        File.WriteAllBytes("label.png", labelData);
    }
}
```

Using this sample label data:

```zpl
^XA

^BY3,2,100
^FO100,750,^BC^FDTEST0123456789^FS

^XZ
```

Actual:
<img src="https://user-images.githubusercontent.com/4907320/174963293-995cfa64-5313-4d69-9da4-de22c130206b.png" width="256" />

Expected (labelary.com reference):
<img src="https://user-images.githubusercontent.com/4907320/174963410-585da1c7-845a-4c73-b782-dc9074b57a4b.png" width="256" />

**Comment by primo-ppcg** [comment]
@tinohager thanks for the quick reply.

I looks like `IncludeLabel` isn't being specified in the various BarcodeElementDrawers:

```csharp
var barcodeElement = new Barcode
{
    BarWidth = barcode.ModuleWidth,
    BackColor = Color.Transparent,
    Height = barcode.Height
};
```

For the example above, I was able to get fairly reasonable results using the following:

```csharp
var barcodeElement = new Barcode {
    BarWidth = barcode.ModuleWidth,
    BackColor = Color.Transparent,
    Height = barcode.Height + 33,
    IncludeLabel = true,
    LabelFont = new Font("DejaVu Sans Mono", 22)
};
```

<img src="https://user-images.githubusercontent.com/4907320/175008957-1237cf79-2f4c-4fdb-97a3-c7bdf5f682b5.png" width="250" />

The appropriate label size and additional height would need to computed at some point before the draw, but I don't see a convenient site. Likely theses would need to be passed as properties of the ZplBarcode.

**Comment by tinohager** [comment]
The problem is that the label can also be above the barcode

```
^XA

^FX Third section with bar code.
^BY5,2,270
^FO100,550
^B1N,N,158,Y,Y
^FD12345678^FS

^XZ
```

**Comment by ingfdoaguirre** [comment]
you need to get the text width and define offset in X and Y depending on the case:

```
//Define offset in X and Y
offsetX := x + (barcodeWidth-textWidth)/2
offsetY := y + barcodeHeight

//If text needed to be above barcode, recalculate offsets
if PrintInterpretationLineAboveCode {
	offsetX = x + (barcodeWidth-textWidth)/2
	offsetY = y - fontHeight
}
```

This is a pseudocode, but It will work, I have ported the viewer to Golang and draw the text correctly

**Comment by tinohager** [comment]
Have you already built your port on the Protocol project?
https://github.com/BinaryKits/BinaryKits.Zpl/discussions/120

**Comment by primo-ppcg** [comment]
> The problem is that the label can also be above the barcode

You can also set the `LabelPosition` of the `BarcodeLib.Barcode` to `TOPCENTER`, e.g.:

```csharp
var barcodeElement = new Barcode {
    BarWidth = barcode.ModuleWidth,
    BackColor = Color.Transparent,
    Height = barcode.Height + barcode.LabelHeight,
    IncludeLabel = true,
    LabelPosition = LabelPositions.TOPCENTER,
    LabelFont = barcode.LabelFont
};
```

<img src="https://user-images.githubusercontent.com/4907320/175220230-33023cb1-b3bd-4f14-b3ca-8b24d6d676aa.png" width="250" />

**Comment by primo-ppcg** [comment]
> @primo-ppcg Can you create a pull request?

The one bit of difficulty that I'm having is that `BarcodeLib` takes a `System.Drawing.Font` rather than a `SKFont`. I'm not quite sure how to deal with that in an elegant way.

**Comment by YipingRuan** [comment]
@primo-ppcg 
Is this fixed by https://github.com/BinaryKits/BinaryKits.Zpl/pull/122 ?

**Comment by primo-ppcg** [comment]
> @primo-ppcg Is this fixed by #122 ?

Mode `A` is complete, mode `N` is partially supported. I intend to add support for mode `D`, and more complete support for mode `N`  - as much as can be with the BarcodeLib library.
